packages/serverpod_shared/lib/src/config.dart

```diff
@@ -74,14 +74,30 @@ class ServerpodConfig {
 
     var doc = loadYaml(data);
 
+    /// Get api server setup. This field cannot be null, so if the
+    /// configuration is missing, create a default configuration.
     assert(doc['apiServer'] is Map, 'apiServer is missing in confing');
-    var apiServer = ServerConfig._fromJson(doc['apiServer']);
-
+    Map apiSetup = doc['apiServer'] ?? {};
+    var apiServer = apiSetup.isNotEmpty
```

Comment:
Suggestion: Make an explicit nullcheck here and throw if the `apiServer` key is missing, as it is a required field. Remove the assert.